### PR TITLE
Initial Python code structure using Poetry

### PR DESCRIPTION
**Pull request by andriika**
Initial Python code structure.

## Proposal for Poetry

This PR advocates for adopting Poetry over Bazel for Python packaging and dependency management. Recent trends and developer feedback show that Poetry is more favored within the Python ecosystem for modern projects.

## A note on gRPC

However, Bazel should still be used to generate gRPC proto stubs and store the output in the `python/michelangelo/proto` directory. We can then add these stubs from (`python/michelangelo/**/*_pb2.py`) to `.gitignore` and configure the packaging script to run Bazel proto generation before running `poetry build`.

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Development
+
+Preprequisites:
+
+- Poetry: https://python-poetry.org
+- Python 3.9
```
Comment (badcount) [inline python/README.md:15]: Are we still using 3.9? shouldn't be 3.10 more popular?

Comment (andriika) [reply python/README.md:15]: Think of it as the minimal Python version we will support, assuming later Python versions are backward-compatible with 3.9.

File: python/poetry.lock
Comment (sallycr) [inline python/poetry.lock:1]: @andriikali delete this file?

**Comment by ying-zheng** [comment]
why gRPC proto stubs still need to use bazel?

File: python/README.md (hunk 1 above)
Comment (mincai) [reply python/README.md:15]: We should start with the latest python version so that we can use latest pytorch etc.

File: python/poetry.lock
Comment (andriika) [reply python/poetry.lock:1]: Usually, it's a good practice to keep a lock file checked in.

File: python/examples/bert_cola/__main__.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import michelangelo as ma
+
+
+@ma.task(config="test")
```
Comment (mincai) [inline python/examples/bert_cola/__main__.py]: shouldn't this be `@uniflow.task` ?

**Review by mincai — APPROVED** [review]

File: python/README.md (hunk 1 above)
Comment (andriika) [reply python/README.md:15]: The latest torch (2.5.1) works for Python 3.9, according to their official guide: https://pytorch.org/get-started/locally/
> It is recommended that you use Python 3.9 - 3.12.

We should agree on the minimum supported Python version and dev/test things against it first. 3.9 is a good minimum version to stick to.

File: python/examples/bert_cola/__main__.py (hunk 2 above)
Comment (sallycr) [reply python/examples/bert_cola/__main__.py]: @mincai I think we would go with ma? Do you think we show the product as uniflow or ma?

File: python/poetry.lock
Comment (sallycr) [reply python/poetry.lock:1]: ack. thanks.

**Review by sishilong — APPROVED** [review]

**Comment by andriika** [comment]
> why gRPC proto stubs still need to use bazel?

@ying-zheng, as you suggested in our recent conversation, I will look into https://github.com/fedej/poetry-grpc-plugin to see if it can be used as an alternative to Bazel.

File: python/examples/bert_cola/__main__.py (hunk 2 above)
Comment (andriika) [reply python/examples/bert_cola/__main__.py]: Changed to:
```
import michelangelo.uniflow as uniflow

@uniflow.task( ... )
```

**Pull request merged**